### ISSUE TITLE
Problems with socat/npiperelay when used on WSL1

### ISSUE THREAD

**Issue by Firehawke** [rupor-github/wsl-ssh-agent#14]
I'm on W10 19041.207 and using your .bashrc trick for WSL2 gives me:
```
Cannot open netlink socket: Protocol not supported
```

There are nine of these lines, and obviously there's no socat to npiperelay.exe link. Any ideas on how I can get around this?

**Comment by rupor-github** [comment]
I am not sure I understand your question - I am on W10 19041.207 and socat/npiperelay combo works perfectly. What exactly are you asking?

**Comment by Firehawke** [comment]
Just erased a whole bunch of debug stuff I was going to include. Seems like it's a problem with the kernel provided by Microsoft, as `ss -a` spams a lot of `Cannot open netlink socket: Protocol not supported` messages. That's probably killing socat's ability to talk to npiperelay.

If I get this figured out, I'll report back so you can add it to the instructions since it definitely seems broken by whatever Microsoft's new default is.

**Comment by rupor-github** [comment]
In my case no such lines at all... How about using `netstat -a` instead?

**Comment by Firehawke** [comment]
Looks like the real problem is that it migrated it back down to WSL1 when I did a distro reset, which is counter-intuitive. I'd checked it before doing the reset and confirmed it was on 2, didn't think to look again after the reset. If this fixes it, I'll suggest adding that error message as a note you should do ``wsl.exe -l -all -v`` to make sure you're REALLY on WSL2.

I'm almost certain it'll work after I force it to upgrade.

**Comment by rupor-github** [comment]
I will certainly mention this in the readme :) My .zshrc has a check:
```
if [ -z "${ISWSL}" ]; then
    if [  $(uname -a | grep -c "Microsoft") -eq 1 ]; then
        export ISWSL=1 # WSL 1
    elif [ $(uname -a | grep -c "microsoft") -eq 1 ]; then
        export ISWSL=2 # WSL 2
    else
        export ISWSL=0
    fi
fi
```
It is ugly but it is works. Depending on the result I will either use socket provided by wsl-ssh-agent.exe from Windows end (WSL1) or use socat/npiperelay.exe (WSL2). This way I do not have to remember anything.

**Comment by Firehawke** [comment]
Okay, just wanted to let you know that after rebuilding my Debian install (the upgrade actually trashed it somehow on the first try, forcing me to remove it and wsl --import the tar.gz backup I make every Friday) I managed to get everything working finally. Also, thanks for the detection method-- I've got my .zshrc in a private Git repo and use it on multiple machines, and it'll be helpful to set things up that way.

All in all, half user error, half "It's broken in a very Microsoft way" but it's now working.
